File: programmers/Level. 2/기능개발.py
# 프로그래머스 Lv.2 기능 개발

def solution(progresses, speeds):
    answer = [] # 정답 리스트
    
    # 작업 리스트가 채워져있을 경우 반복
    while progresses :
        progresses = [x + y for x, y in zip(progresses, speeds)] # 두 리스트를 합함
        
        # 맨 앞 작업이 완료되었을 경우
        if progresses[0] >= 100 :
            answer.append(0) # answer 변수 추가
            while progresses and progresses[0] >= 100 : # 작업 리스트가 채워져있고, 맨 앞 작업이 완료된 이상 반복
                answer[-1] += 1
                progresses.pop(0) # 맨 앞 작업 제거
                speeds.pop(0) # 해당 작업의 속도 리스트도 한 칸 제거
    
    return answer

File: programmers/Level. 2/기능개발.cpp
// 프로그래머스 Lv.2 기능 개발

#include <string>
#include <vector>

using namespace std;

struct Process
{
    int progress;
    int speed;
};

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    
    // 1. Process 배열 생성
    vector<Process> p;
    
    for (int i = 0; i < progresses.size(); i++) {
        Process process;
        process.progress = progresses[i];
        process.speed = speeds[i];
        p.push_back(process);
    }
    
    // 2. 반복문 안에서 Process의 progress에 speed를 증감
    while (!p.empty()) {
        for (int i = 0; i < p.size(); i++)
            p[i].progress += p[i].speed;
        
        // 3. 가장 앞의 Process의 progress가 100 이상이라면, 앞에서부터 프로세스 종료
        if (p.front().progress >= 100) {
            int count = 0;
            while (!p.empty() && p.front().progress >= 100) {
                p.erase(p.begin());
                count++;
            }
            answer.push_back(count);
        }
    }
    
    return answer;
}